### PR TITLE
improvement: ZENKO-2278 count items code reorg

### PR DESCRIPTION
**Pull request by alexanderchan-scality**

### CODE REVIEW

**Review by tmacro — APPROVED** [review]

**Review by JianqinWang — APPROVED** [review]

File: lib/storage/metadata/mongoclient/MongoClientInterface.js [hunk 1]
```diff
@@ -1283,42 +1278,50 @@ class MongoClientInterface {
...
-                    }
-                    return done();
-                });
-            }, err => {
-                if (err) {
-                    return cb(err);
-                }
-                // save to infostore
-                return this.updateCountItems(store, log, err => {
+            let concurrentCursors = CONCURRENT_CURSORS;
```
Comment (rachedbenmustapha) [inline lib/storage/metadata/mongoclient/MongoClientInterface.js]: This seems to be in the way, why not in the constructor or setup method?

File: lib/storage/metadata/mongoclient/MongoClientInterface.js [hunk 2]
```diff
@@ -1415,211 +1417,136 @@ class MongoClientInterface {
...
-        const mstFilter = {
-            '_id': { $regex: /^[^\0]+$/ },
-            'value.versionId': { $exists: true },
-        };
-        const verFilter = { _id: { $regex: /\0/ } };
-        const nullFilter = {
-            '_id': { $regex: /^[^\0]+$/ },
-            'value.versionId': { $exists: false },
+        const cursor = c.find({}, {
+            sort: { _id: 1 },
```
Comment (rachedbenmustapha) [inline lib/storage/metadata/mongoclient/MongoClientInterface.js]: Given that we handle items as they come, is the sort useful? Does it incur a penalty?

File: lib/storage/metadata/mongoclient/MongoClientInterface.js [hunk 3]
```diff
@@ -1415,211 +1417,136 @@ class MongoClientInterface {
...
-            return cb(null, count);
-        });
+    /**
+     * @param{object} entry -
+     * @param{string} entry._id -
+     * @param{object} entry.value -
+     * @param{Date} cmpDate -
+     * @returns{boolean} stalled -
+     */
+    _processEntryStalled(entry, cmpDate) {
```
Comment (rachedbenmustapha) [inline lib/storage/metadata/mongoclient/MongoClientInterface.js]: This is a test, so maybe call it `isReplicationEntryStalled`

File: lib/storage/metadata/mongoclient/MongoClientInterface.js (hunk 2 above)
Comment (alexanderchan-scality) [reply lib/storage/metadata/mongoclient/MongoClientInterface.js]: you are correct. order doesn't matter in the case of count items. the sort can be removed.

**Review by nicolas2bert — APPROVED** [review]

**Review by rachedbenmustapha — APPROVED** [review]

**Comment by alexanderchan-scality** [comment]
rebased/squashed

**Pull request merged**